src/Cli/dotnet/commands/dotnet-new/NewCommandParser.cs

```diff
@@ -60,12 +81,37 @@ static CliTemplateEngineHost GetEngineHost(ParseResult parseResult)
                     || Env.GetEnvironmentVariableAsBool(EnableProjectContextEvaluationEnvVarName);
                 FileInfo? projectPath = parseResult.GetValueForOption(SharedOptions.ProjectPathOption);
                 FileInfo? outputPath = parseResult.GetValueForOption(SharedOptions.OutputOption);
+                
+                //--help doesn't take into account default value
+                // therefore manual handling
+                OptionResult? verbosityOptionResult = parseResult.FindResultFor(s_verbosityOption);
+                VerbosityOptions verbosityOption = verbosityOptionResult == null ? DefaultVerbosity : verbosityOptionResult.GetValueOrDefault<VerbosityOptions>();
```

Comment:
report to System.CommandLine - default value for options are not applied when `--help` is used.